src/interpreter/type_cast.rs

```diff
@@ -0,0 +1,224 @@
+use crate::interpreter::types::{Numeric, PactType, StringLike};
+use core::any::Any;
+use core::convert::TryFrom;
+use std::string::String;
+use std::vec::Vec;
+
+#[derive(Debug, PartialEq)]
+pub enum PactConversionErr {
+    Overflow,
+    UnknownType,
+}
+
+/// A catch-all conversion trait which tries to turn any given `value` into the implementing type
+pub trait AnyTryInto<'a>: Sized {
+    fn any_try_into(value: &'a dyn Any) -> Result<Self, PactConversionErr>;
+}
+
+/// AnyTryInto implementation for PactType
+impl<'a> AnyTryInto<'a> for PactType<'a> {
+    fn any_try_into(value: &'a dyn Any) -> Result<PactType<'a>, PactConversionErr> {
+        // TODO: refactor the below repetiion using macros
+        // Unsigned integer type casting into PactType
+        if let Some(number) = value.downcast_ref::<u8>() {
+            if let Ok(n) = u64::try_from(*number) {
+                return Ok(PactType::Numeric(Numeric(n)));
+            }
+        }
+        if let Some(number) = value.downcast_ref::<u16>() {
+            if let Ok(n) = u64::try_from(*number) {
+                return Ok(PactType::Numeric(Numeric(n)));
+            }
+        }
+        if let Some(number) = value.downcast_ref::<u32>() {
+            if let Ok(n) = u64::try_from(*number) {
+                return Ok(PactType::Numeric(Numeric(n)));
+            }
+        }
+        if let Some(number) = value.downcast_ref::<u64>() {
+            return Ok(PactType::Numeric(Numeric(*number)));
+        }
+        if let Some(number) = value.downcast_ref::<u128>() {
+            if *number > core::u64::MAX as u128 {
+                return Err(PactConversionErr::Overflow);
+            }
+            if let Ok(n) = u64::try_from(*number) {
```

Comment:
The `try_from` impl will check the overflow condition so no need to check here. Can map the error though into PactConversionError e.g. `try_from(*number).map_err(|_| PactConversionErr::Overflow)?`